File: Machine Learning/softmax/submission-0.py
import numpy as np
from numpy.typing import NDArray


class Solution:

    def softmax(self, z: NDArray[np.float64]) -> NDArray[np.float64]:
        # z is a 1D NumPy array of logits
        # Hint: subtract max(z) for numerical stability before computing exp
        # return np.round(your_answer, 4)
        maxi=np.max(z)
        n=len(z)
        ans=[]
        for i in range(n):
            value = z[i]-maxi 
            one = np.exp(value)
            second=0.0
            for j in range(n):
                new_value=z[j]-maxi 
                second = second + np.exp(new_value)
            
            app=np.round(one/second,4) 

            ans.append(app)
        
        return ans

File: Data Structures & Algorithms/anagram-groups/submission-0.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        map<string,vector<string>> mp;
        int n=strs.size();
        for(int i=0;i<n;i++)
        {
            string val=strs[i];
            sort(val.begin(),val.end());
            mp[val].push_back(strs[i]);
        }
        vector<vector<string>> ans;
        for(auto &e:mp)
        {
            ans.push_back(e.second);
        }
        return ans;
    }

};
